### DISCUSSION TITLE
Grid auto height behaving differently between Windows and Android

### DISCUSSION THREAD

**Discussion by MustacheEmperor** [dotnet/maui#14505] [Q&A]
Hi, I'm trying to sort out if I'm doing something wrong or if this should be filed as an issue.

I'm working on a xaml layout that displays a ListView of ViewCells each consisting of an image with text labels displayed underneath. My intent is a long feed of images that horizontally expand to the width of the screen and aspect-fill vertically without cropping. 

The below code displays as described on Windows, but on Android the images are all cropped to thin rectangles and appear to be covering up the text. I am new to Maui - is this my mistake, or an issue? Even setting a HeightRequest does not alter the actual pixel height of the image, it just blows it up within the cropped area. 

Using a VerticalStackLayout instead of a Grid manifests exactly the same behavior - the layout looks as I described above on Windows, on Android the images are the full width but cropped very short. 

```
    <ListView x:Name="Trending"
                ItemsSource="{Binding VideosList}"
                ItemSelected="TrendingOnItemSelected"
                Margin="20"
                >
        <ListView.ItemTemplate>
            <DataTemplate>
                <ViewCell>
                    <Grid>
                        <Grid.RowDefinitions >
                            <RowDefinition Height="Auto"/>
                            <RowDefinition />
                            <RowDefinition />
                        </Grid.RowDefinitions>
                        <Grid.ColumnDefinitions>
                            <ColumnDefinition Width="*"/>
                        </Grid.ColumnDefinitions>
                        <Image
                            Grid.Row="0"
                            Source="{Binding ImagePath}"
                            Aspect="AspectFill"    
                            >
                        </Image>
                        <Label Grid.Row="1" Text="{Binding Name}"></Label>
                        <Label Grid.Row="2" Text="{Binding Description}"></Label>
                    </Grid>
                </ViewCell>
            </DataTemplate>
        </ListView.ItemTemplate>
    </ListView>
```

**Comment by hartez** [comment]
> aspect-fill vertically without cropping

I think `AspectFill` may be the opposite of what you want - [from the docs](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/image?view=net-maui-7.0#control-image-scaling): "_clips_ the image so that it fills the display area while preserving the aspect ratio".

If you don't want any part of the image to be cut off, you probably want `AspectFit`. 

Also, it might help us understand the problem better if you could post some screenshots of what you're seeing on Android vs Windows.

**Reply by MustacheEmperor** [reply]
Thanks for the assistance. I changed the Image aspect to AspectFit for these screenshots, although it doesn't look to have had an effect on either layout. 

Below are screenshots of the app running on Windows 10 at a couple different aspect ratios and demonstrating the intended behavior. Also included is one screenshot of the app running on an emulated Pixel 5 displaying the layout issue I described above. 
![windows_screen_1](https://user-images.githubusercontent.com/10985571/231306030-00d644ba-3732-4469-aa17-4bf11c8ea325.jpg)
![windows_screen_2](https://user-images.githubusercontent.com/10985571/231306034-c827389b-4765-4125-a2c9-ab78e01e02d8.jpg)
![android_pixel5](https://user-images.githubusercontent.com/10985571/231306059-f506a583-4a2b-428b-b828-7870d85acd70.jpg)

**Reply by MustacheEmperor** [reply]
Small correction, those screenshots were actually taken with the VerticalStackLayout. However, the behavior manifests almost exactly the same. Below are screenshots of a grid layout. 

![image](https://user-images.githubusercontent.com/10985571/231306682-15460bac-d28a-4595-9a58-0d57030733d9.png)
![image](https://user-images.githubusercontent.com/10985571/231306735-aeab80f0-b490-44b7-b5f8-1193930864be.png)

**Reply by MustacheEmperor** [reply]
I've posted a repo demonstrating the behavior here: https://github.com/MustacheEmperor/MauiAndroidLayoutIssue

**Reply by MustacheEmperor** [reply]
I found this issue which seems related: https://github.com/dotnet/maui/issues/6651

Setting HasUnevenRows=True results in behavior on Android matching Windows.

**Reply by hartez** [reply]
I've converted this to its own issue so we can track it and look into it.
